### ISSUE TITLE
Tflite model provides good accuracy on edge device but not in pythonic Interpreter

### ISSUE THREAD

**Issue by muck27** [tensorflow/tensorflow#38723]
OS system: linux (mintos)
Graphics Card: Nvidia 1660 ti MaxQ
System: Rog Zephyrus 

Problem:
I have converted my model to tflite and installed it on android device. The performance and accuracy seems to be good enough for a started point but when I use the tflite model in the tflite interpreter I get completly bad results. 
I would like to know what I have done wrong.

This is the code to load tflite and the interpreter

[tflite_detect.txt](https://github.com/tensorflow/tensorflow/files/4506390/tflite_detect.txt)

**Comment by Saduf2019** [comment]
@muck27 
Please provide details about  your TensorFlow version. 
f you are unclear what to include see the issue template displayed in [the Github new issue template](https://github.com/tensorflow/tensorflow/issues/new).

We ask for this in the issue submission template, because it is really difficult to help without that information. Thanks!

**Comment by muck27** [comment]
Hello @Saduf2019 ..
TF model: 2.1.0
method for conversion:

1. Convert graph to tflite.graph 

$python research/object_detection/export_tflite_ssd_graph.py --pipeline_config_path=$CONFIG_FILE --trained_checkpoint_prefix=$CHECKPOINT_PATH --output_directory=$OUTPUT_DIR --add_postprocessing_op=true


2. Convert the tflite graph to tflite model:

tflite_convert --output_file=./tflite/detect.tflite --graph_def_file=/models/tflite/tflite_graph.pb --input_arrays=normalized_input_image_tensor --output_arrays='TFLite_Detection_PostProcess','TFLite_Detection_PostProcess:1','TFLite_Detection_PostProcess:2','TFLite_Detection_PostProcess:3' --input_shape=1,300,300,3 --allow_custom_ops

**Comment by thaink** [comment]
@muck27,
If your model are train using coco dataset, I would recommend to check the accuracy using that dataset first before using images from your camera. 
to check it on PC, you can try with https://github.com/mlperf/mobile_app/tree/master/cpp/binary:
bazel run  -c opt --cxxopt='--std=c++17' -- //cpp/binary:main TFLITE COCO \
third_party/mlperf/mobile_app/cpp/binary:main \
--mode=SubmissionRun \
--model_file=<path to your model> \
--images_directory=<path to coco image dir> \
--groundtruth_file=<ground truth file> \
--num_threads=4 --num_classes=91 --offset=1 \
--delegate=None \
--output_dir=/<output dir>
The guide to download the coco dataset can be found at https://github.com/mlperf/mobile_app/tree/master/cpp/datasets

**Comment by muck27** [comment]
@thaink 

I am building a custom object detector. So what should my groundtruth file be like? Does it need to be in a specific format like pascal or in tfrecord format?

**Comment by thaink** [comment]
@muck27, Do you mean you are using your own dataset?

**Comment by muck27** [comment]
@thaink yes I am

**Comment by thaink** [comment]
@muck27 My main suspect is the pre-processing code.
Could you check the pre-processing that you use on Android and in the python code is the same?

**Comment by muck27** [comment]
@thaink even I feel the preprocessing part is giving wrong results. 
I have trained the model with input of 1200x1200.  I am attaching the android preprocessing part and the python code for interpretation. 

[tflite_detect.txt](https://github.com/tensorflow/tensorflow/files/4523345/tflite_detect.txt)
![android](https://user-images.githubusercontent.com/48056260/80115188-0eef9580-85a2-11ea-9b4d-974de59e7747.png)

**Comment by thaink** [comment]
my guess is you are missing the normalization part here
https://github.com/tensorflow/examples/blob/318936de60224383901e5ef60a39799629a5df8d/lite/examples/object_detection/android/app/src/main/java/org/tensorflow/lite/examples/detection/tflite/TFLiteObjectDetectionAPIModel.java#L162
But if you trained the model, you only need to make sure the way you preprocess the image is the same as the way you trained it.

**Comment by muck27** [comment]
I will try normalizing the image as a preprocessing part before sending it to the network then

**Comment by Saduf2019** [comment]
Moving this to resolved status with confirmation.
